imp_act/environments/road_env.py

```diff
@@ -121,9 +145,26 @@ def get_initial_state(self):
         self.state = self.initial_state
         self.observation = self.random_generator.choice(
             np.arange(self.number_of_states),
-            p=self.observation_tables[0][self.state],
+            p=self.observation_tables[self.ACTION_DO_NOTHING][self.state],
         )
 
+    def get_action_reward(self, action):
+        reward = 0
+        if action == self.ACTION_REPLACE:
+            forced_repair_interest = (
+                self.forced_repair_interest_rate**self.forced_repair_interest_counter
+            )
+            reward = (
+                self.state_action_reward[action][self.state] * forced_repair_interest
+            )
+            self.forced_repair_interest_counter = 0
+        else:
+            reward = self.state_action_reward[action][self.state]
+        return reward
+
+    def get_terminal_reward(self):
```

Comment:
Are we sure we would like to charge terminal costs based on beliefs? I suggested this option, but is this okay with our reward formulation? Let's perhaps all agree.